File: app/src/main/java/com/example/bg6tracker/MainActivity.java
package com.example.bg6tracker;

import android.app.Activity;
import android.content.Intent;
import android.media.projection.MediaProjectionManager;
import android.os.Bundle;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

public class MainActivity extends Activity {

    private static final int REQUEST_CAPTURE = 1001;

    private MediaProjectionManager projectionManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        projectionManager =
                (MediaProjectionManager)
                        getSystemService(
                                MEDIA_PROJECTION_SERVICE
                        );

        LinearLayout layout =
                new LinearLayout(this);

        layout.setOrientation(
                LinearLayout.VERTICAL
        );

        layout.setPadding(
                40,
                60,
                40,
                40
        );

        TextView title =
                new TextView(this);

        title.setText(
                "BG6 Screen Tracker"
        );

        title.setTextSize(24);

        Button start =
                new Button(this);

        start.setText(
                "Bắt đầu thu màn hình"
        );

        start.setOnClickListener(
                v -> requestCapture()
        );

        layout.addView(title);
        layout.addView(start);

        setContentView(layout);
    }

    private void requestCapture() {

        Intent intent =
                projectionManager
                        .createScreenCaptureIntent();

        startActivityForResult(
                intent,
                REQUEST_CAPTURE
        );
    }

    @Override
    protected void onActivityResult(
            int requestCode,
            int resultCode,
            Intent data) {

        super.onActivityResult(
                requestCode,
                resultCode,
                data
        );

        if (requestCode != REQUEST_CAPTURE) {
            return;
        }

        if (resultCode != RESULT_OK ||
                data == null) {
            return;
        }

        Intent service =
                new Intent(
                        this,
                        CaptureService.class
                );

        service.putExtra(
                CaptureService.EXTRA_RESULT_CODE,
                resultCode
        );

        service.putExtra(
                CaptureService.EXTRA_RESULT_DATA,
                data
        );

        if (android.os.Build.VERSION.SDK_INT >= 26) {
            startForegroundService(service);
        } else {
            startService(service);
        }
    }
}

File: app/src/main/java/com/example/bg6tracker/MainActivity.kt
package com.example.bg6tracker

import android.app.Activity
import android.content.Intent
import android.media.projection.MediaProjectionManager
import android.os.Bundle
import android.widget.Button
import android.widget.TextView

class MainActivity : Activity() {

    companion object {

        private const val REQUEST_CAPTURE = 500
    }

    private lateinit var projectionManager:
        MediaProjectionManager

    private lateinit var status:
        TextView

    override fun onCreate(
        savedInstanceState: Bundle?
    ) {

        super.onCreate(
            savedInstanceState
        )

        setContentView(
            R.layout.activity_main
        )

        status =
            findViewById(
                R.id.statusText
            )

        val start =
            findViewById<Button>(
                R.id.startButton
            )

        val stop =
            findViewById<Button>(
                R.id.stopButton
            )

        projectionManager =
            getSystemService(
                MEDIA_PROJECTION_SERVICE
            ) as MediaProjectionManager

        start.setOnClickListener {

            val intent =
                projectionManager
                    .createScreenCaptureIntent()

            startActivityForResult(
                intent,
                REQUEST_CAPTURE
            )
        }

        stop.setOnClickListener {

            stopService(
                Intent(
                    this,
                    CaptureService::class.java
                )
            )

            status.text =
                "Status: Stopped"

            start.isEnabled = true
            stop.isEnabled = false
        }
    }

    override fun onActivityResult(
        requestCode: Int,
        resultCode: Int,
        data: Intent?
    ) {

        super.onActivityResult(
            requestCode,
            resultCode,
            data
        )

        if (
            requestCode ==
            REQUEST_CAPTURE &&
            resultCode ==
            RESULT_OK &&
            data != null
        ) {

            val serviceIntent =
                Intent(
                    this,
                    CaptureService::class.java
                )

            serviceIntent.action =
                CaptureService.ACTION_START

            serviceIntent.putExtra(
                CaptureService.EXTRA_RESULT_CODE,
                resultCode
            )

            serviceIntent.putExtra(
                CaptureService.EXTRA_DATA,
                data
            )

            startForegroundService(
                serviceIntent
            )

            status.text =
                "Status: Capturing"

            findViewById<Button>(
                R.id.startButton
            ).isEnabled = false

            findViewById<Button>(
                R.id.stopButton
            ).isEnabled = true
        }
    }
}
